concordium-consensus/src/Concordium/KonsensusV1/Scheduler.hs

```diff
@@ -231,6 +231,11 @@ doEpochTransition True epochDuration theState0 = do
                 -- This is the start of the last epoch of a payday, so take a baker snapshot.
                 let epochEnd = newSeedState ^. triggerBlockTime
                 let av = accountVersionFor (demoteProtocolVersion (protocolVersion @(MPV m)))
+
+                -- TODO (drsk) suspend validators that are primed for suspension and whose 
+                -- missed rounds are above the threshold set by the score chain parameter.
+                -- Clear all active bakers from being primed for suspension
```

Comment:
I think this is good and in line with the proposal.

I mentioned that we should add special transaction outcomes for suspending accounts. My main concern with this is that the special transaction outcomes are treated as happening after the regular transactions. For instance, if an account receives a reward in a block (which is in a special transaction outcome), that reward cannot be spent in the same block by a regular transaction. This means that if we have a special transaction outcome suspending a validator and a regular transaction resuming the validator in the same block, then we have to be a bit careful about handling this. In particular:
- The validator should be considered suspended for the purposes of calculating the next epoch bakers.
- There will be a transaction outcome indicating the validator was resumed manually.
- There should be a special transaction outcome indicating the validator was suspended automatically.
- The interpretation of special transaction outcomes happening after the regular transactions means we should actually do the suspension when executing the block epilogue.

Reply:
I implemented this as discussed above (I think). @td202 could you take a look at `processSuspension` to tell me that this is the right direction?